### PR TITLE
Separate enablePreview setting for search view

### LINKED ISSUE #60131: Enable separeate preview modes for Search & Replace results
There is a setting called `"workbench.editor.enablePreview"`, which I currently have set to `false`, mainly because I use my keyboard arrows to navigate through files in the file explorer, and it annoys me that when I open a file pressing Enter, it opens it in preview mode (I would have to double click on it with my mouse to open it and keep it, but as I said I am barely using my mouse).

However, when I am watching my Search & Replace results, I don't like that every file with a search result opens when I navigate through it. After I finish my search, I end up with a bunch of open editors that I am not going to use anymore. Here is where I would like to use the preview mode.

To summarize, I would like to have the preview modes separately for each context: something like  `"workbench.editor.enablePreviewFromExplorer"` and `"workbench.editor.enablePreviewFromSearch"`. There is currently a `"workbench.editor.enablePreviewFromQuickOpen"`, which I find very useful.

### PR DESCRIPTION
**Pull request by kyleshoe2**
This PR fixes #60131.

I also noticed that enablePreviewFromQuickOpen didn't operate as expected if it was set to true and enablePreview was set to false, i.e. opening an editor from Quick Open would open it as a non-preview editor when I would expect it to be a preview editor. This is due to `!this.accessor.partOptions.enablePreview` being used in the `pinned` argument in the `openEditorOptions` in editorGroupView.ts. Since several other editors like Settings and the Welcome page depend on this behavior for opening in preview/non-preview mode, I opted to make an editor option to override the setting.

### CODE REVIEW

**Comment by jbracer19** [comment]
Workbench needs some clarity with this.
Black / White would seem simply at local level, can i get that?

On Oct 20, 2019, at 10:33 PM, Kyle Shoemaker <notifications@github.com> wrote:

﻿

This PR fixes #60131<https://github.com/microsoft/vscode/issues/60131>.

I also noticed that enablePreviewFromQuickOpen didn't operate as expected if it was set to true and enablePreview was set to false, i.e. opening an editor from Quick Open would open it as a non-preview editor when I would expect it to be a preview editor. This is due to !this.accessor.partOptions.enablePreview being used in the pinned argument in the openEditorOptions in editorGroupView.ts. Since several other editors like Settings and the Welcome page depend on this behavior for opening in preview/non-preview mode, I opted to make an editor option to override the setting.

________________________________
You can view, comment on, or merge this pull request online at:

  https://github.com/microsoft/vscode/pull/82948

Commit Summary

  *   Add enablePreviewFromSearch setting
  *   Make search view set pinned based on enablePreviewFromSearch
  *   Create enablePreview option in EditorOptions to override enablePreview setting
  *   Use enablePreview editor option to enable/disable preview editor based on correct setting for Quick open and search views
  *   Merge branch 'master' into separatePreviewForSearch

File Changes

  *   M src/vs/platform/editor/common/editor.ts<https://github.com/microsoft/vscode/pull/82948/files#diff-0> (5)
  *   M src/vs/workbench/browser/parts/editor/editorGroupView.ts<https://github.com/microsoft/vscode/pull/82948/files#diff-1> (11)
  *   M src/vs/workbench/browser/parts/quickopen/quickOpenController.ts<https://github.com/microsoft/vscode/pull/82948/files#diff-2> (7)
  *   M src/vs/workbench/browser/workbench.contribution.ts<https://github.com/microsoft/vscode/pull/82948/files#diff-3> (5)
  *   M src/vs/workbench/common/editor.ts<https://github.com/microsoft/vscode/pull/82948/files#diff-4> (10)
  *   M src/vs/workbench/contrib/search/browser/searchView.ts<https://github.com/microsoft/vscode/pull/82948/files#diff-5> (6)

Patch Links:

  *   https://github.com/microsoft/vscode/pull/82948.patch
  *   https://github.com/microsoft/vscode/pull/82948.diff

**Comment by kwshoemaker2** [comment]
@jbracer19 I'm not sure what you mean. What does workbench need clarity on?

**Comment by jbracer19** [comment]
The file showing original desktop browser install/responsible parties and the distribution of well me , I’ve seen it all but simply can’t extract it so that it’s easily  understandable to someone that doesn’t do code. 903 915. 9134



On Oct 20, 2019, at 11:08 PM, Kyle Shoemaker <notifications@github.com> wrote:

﻿

@jbracer19<https://github.com/jbracer19> I'm not sure what you mean. What does workbench need clarity on?

File: src/vs/platform/editor/common/editor.ts [hunk 1]
```diff
@@ -170,6 +170,11 @@ export interface IEditorOptions {
 	 */
 	readonly pinned?: boolean;
 
+	/**
+	 * Will cause the editor to open as a preview editor.
+	 */
+	readonly enablePreview?: boolean;
```
Comment (roblourens) [inline src/vs/platform/editor/common/editor.ts]: This is the same as "pinned"

**Review by roblourens — CHANGES_REQUESTED** [review]
Please don't change the behavior of the other "enable preview" settings - I think it is intentional that "Enable Preview" is a master switch to disable is globally, where "Enable Preview From Quick Open" just lets you disable it on quick open (but not override the global setting)

**Comment by kwshoemaker2** [comment]
Oh sorry I think I got mixed up initially. Looking at the original issue they were asking for an enablePreviewFromSearch AND enablePreviewFromExplorer, which would affect the editor when opened from the respective view.

I think I read your comment at the bottom saying that you’d accept PRs for the enablePreviewFromSearch and took that to mean only adding that setting. Do you want me to add enablePreviewFromExplorer as well?

**Comment by roblourens** [comment]
Now I'm not sure. I added a comment to the original issue to continue the discussion.

**Pull request closed without merging**